### PR TITLE
Fix order parameters and add decimal to to_string

### LINKED ISSUE #23612: CLN: Consistency of arguments in to_html and to_string
The methods `to_html` and `to_string` share most of the parameters, together with `DataFrameFormatter`, but there are some differences in the order, and in some parameters that probably should be the same.

Those are the common parameters:
```
buf=None,
columns=None,
col_space=None,
header=True,
index=True,
na_rep='NaN',
formatters=None,
float_format=None,
sparsify=None,
index_names=True,
justify=None,
max_rows=None,
max_cols=None,
show_dimensions=False
```

But after `formatters` they are in different order, which adds complexity (for example in reusing docstrings, or in switching from one function to another when using positional arguments).

Then, those are the parameters that are not shared among the 3:
```
DataFrameFormatter
--
frame,  # (positional, at the beginning)
line_width=None,
decimal='.',
table_id=None,

to_string
--
line_width=None,

to_html
--
decimal='.',
table_id=None
bold_rows=True,
classes=None,
escape=True,
notebook=False,
border=None,
```

While that looks mostly ok, it feels like probably `to_string` should also have a `decimal` argument (and not sure what `DataFrameFormatter` should have.

What I propose is:
- Reorder the arguments so the common ones are located first, and in the same order across all methods.
- Add `decimal` to `to_string`, and check if any other addition (or deletion) makes sense.
- Add a note in whatsnew, but don't do any deprecation (which would make things quite tricky in our code). Hopefully no users are using positional arguments for the last arguments in a method with around 15.
- Make sure the docstrings have the parameters in the right order.
- Looks like in `pandas/io/formats/format.py` there is a related `docstring_to_string` variable that is not being used, so we should remove it.

@jreback happy to make these changes?

CC @thoo

### PR DESCRIPTION
**Pull request by thoo**
- [x] closes #23612
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry
Add an extra argument:`decimal` to `pandas.DataFrame.to_string`.

### CODE REVIEW

**Comment by datapythonista** [comment]
Looks quite nice.

I think it good be good to have one test to see that `decimal` is working in `to_string`, and then we need to add to the whatsnew two entries, one for the addition of decimal, and another one for the argument resorting.

**Comment by thoo** [comment]
@datapythonista Please let me know the two entries are not in the right place in `whatnews` file.

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -945,6 +946,7 @@ Other API Changes
 - :class:`DateOffset` attribute `_cacheable` and method `_should_cache` have been removed (:issue:`23118`)
 - Comparing :class:`Timedelta` to be less or greater than unknown types now raises a ``TypeError`` instead of returning ``False`` (:issue:`20829`)
 - :meth:`Index.hasnans` and :meth:`Series.hasnans` now always return a python boolean. Previously, a python or a numpy boolean could be returned, depending on circumstances (:issue:`23294`).
+- The order of the arguments of `DataFrame.to_html` and `DataFrame.to_string` is rearranged. (:issue:`23614`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: can you use a `:func:` ref here

Comment (jreback) [reply doc/source/whatsnew/v0.24.0.txt]: say to be consistent with each other

File: pandas/tests/io/formats/test_format.py [hunk 2]
```diff
@@ -1451,6 +1451,11 @@ def test_to_string_format_na(self):
                     '4  4.0     bar')
         assert result == expected
 
+    def test_to_string_decimal(self):
+        df = DataFrame({'A': [6.0, 3.1, 2.2]})
```
Comment (jreback) [inline pandas/tests/io/formats/test_format.py]: can you add the issue number

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by thoo** [comment]
@jreback  All checks are passed. Could you re-review this ? Thanks.

File: doc/source/whatsnew/v0.24.0.txt [hunk 3]
```diff
@@ -20,7 +20,8 @@ New features
 the user to override the engine's default behavior to include or omit the
 dataframe's indexes from the resulting Parquet file. (:issue:`20768`)
 - :meth:`DataFrame.corr` and :meth:`Series.corr` now accept a callable for generic calculation methods of correlation, e.g. histogram intersection (:issue:`22684`)
-
+- :func:`DataFrame.to_string` now accepts ``decimal`` as an argument, allowing
+the user to use a decimal separator. (:issue:`23614`)
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.24.0.txt]: more than `to use` as decimal separator, if I understood correctly, it'll allow the user to specify which decimal separator should be used in the output. Isn't it?

File: pandas/tests/io/formats/test_format.py (hunk 2 above)
Comment (datapythonista) [reply pandas/tests/io/formats/test_format.py]: Small thing, but I think we consistently use the format `GH 23614`. 

@gfyoung I think you changed some in a recent PR, can you confirm?

**Review by datapythonista — COMMENTED** [review]
looks good, just two minor things

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @datapythonista merge when you are satisfied.

File: doc/source/whatsnew/v0.24.0.rst [hunk 4]
```diff
@@ -961,6 +962,7 @@ Other API Changes
 - :class:`DateOffset` attribute `_cacheable` and method `_should_cache` have been removed (:issue:`23118`)
 - Comparing :class:`Timedelta` to be less or greater than unknown types now raises a ``TypeError`` instead of returning ``False`` (:issue:`20829`)
 - :meth:`Index.hasnans` and :meth:`Series.hasnans` now always return a python boolean. Previously, a python or a numpy boolean could be returned, depending on circumstances (:issue:`23294`).
+- :func:The order of the arguments of `DataFrame.to_html` and `DataFrame.to_string` is rearranged to be consistent with each other. (:issue:`23614`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.rst]: this is malformed here, you need the ``:func:`` to go with the DataFrame.to_html and .to_string (separate ones)

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.24.0.rst (hunk 4 above)
Comment (thoo) [reply doc/source/whatsnew/v0.24.0.rst]: Got it. Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**